tests/infrastructure/vm_console_proxy/utils.py

```diff
@@ -66,3 +68,21 @@ def get_vm_console_proxy_resource(resource_kind, namespace=None):
             name=f"{TOKEN_API_VERSION}.{TOKEN_ENDPOINT}" if resource_kind == APIService else VM_CONSOLE_PROXY
         )
     return vm_console_proxy_resource_object
+
+
+def assert_resource_status_and_wait_for_availability(resource: Any, timeout: int) -> None:
+    assert resource.exists, (
+        f"Missing: {resource.kind}/{resource.name} under {getattr(resource, 'namespace', 'cluster')}"
+    )
+
+    if not hasattr(resource, "status"):
+        LOGGER.info(f"Resource {resource.kind} has no status. Skipping availability check.")
+        return
+
+    if getattr(resource.instance.status, "conditions", None):
+        LOGGER.info(f"Checking availability of {resource.kind}")
+        resource.wait_for_condition(
+            condition=Resource.Condition.AVAILABLE,
+            status=Resource.Condition.Status.TRUE,
+            timeout=timeout,
+        )
```

Comment:
If you are making this call, I am assuming status.conditions are supposed to be present? This won't fail if it is not present. That seems like an automation bug to me.